src/Controls/src/Core/Platform/Android/PointerGestureHandler.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using Android.Views;
+using Microsoft.Maui.Controls.Platform.Android;
+using Microsoft.Maui.Graphics;
+using AView = Android.Views.View;
+
+namespace Microsoft.Maui.Controls.Platform
+{
+	internal class PointerGestureHandler : Java.Lang.Object, AView.IOnHoverListener
+	{
+		internal PointerGestureHandler(Func<View> getView, Func<AView> getControl)
+		{
+			GetView = getView;
+			GetControl = getControl;
+			SetupHandlerForPointer();
+		}
+
+		Func<View> GetView { get; }
+		Func<AView> GetControl { get; }
+
+		public bool OnHover(AView control, MotionEvent e)
+		{
+			var view = GetView();
+
+			foreach (var gesture in view.GetCompositeGestureRecognizers())
+			{
+				if (gesture is PointerGestureRecognizer)
+				{
+					var pgr = gesture as PointerGestureRecognizer;
+					switch (e.Action)
+					{
+						case MotionEventActions.HoverEnter:
+							pgr.SendPointerEntered(view, (relativeTo) => CalculatePosition(view, e));
+							break;
+						case MotionEventActions.HoverMove:
+							pgr.SendPointerMoved(view, (relativeTo) => CalculatePosition(view, e));
+							break;
+						case MotionEventActions.HoverExit:
+							pgr.SendPointerExited(view, (relativeTo) => CalculatePosition(view, e));
+							break;
+					}
+				}
+			}
+
+			return false;
+		}
+
+		public void SetupHandlerForPointer()
+		{
+			var view = GetView();
+			if (view == null)
+				return;
+
+			var control = GetControl();
+			if (control == null)
+				return;
+
+			if (HasAnyPointerGestures())
+				GetControl()?.SetOnHoverListener(this);
+
+			return;
+		}
+
+		Point? CalculatePosition(IElement element, MotionEvent e)
```

Comment:
Should we move this to a `MotionEventExtensions` file? That way we aren't duplicating the code between here and `tapGestureHandler` ?